x-pack/plugins/security_solution/public/cloud_security_posture/components/alerts/alerts_preview.tsx

```diff
@@ -82,9 +107,120 @@ export const AlertsPreview = ({
 
   const totalAlertsCount = alertStats.reduce((total, item) => total + item.count, 0);
 
+  const { data } = useMisconfigurationPreview({
```

Comment:
I don't understand why we need all this logic (getting misconfigs, vulnerabilities, riskScore) in the alert preview. From what I see we open the Alerts data grid in the left section of the flyout, this section doesn't have information about risk, misconfigs and vulnerabilities, then why all this data is needed?

Reply:
We need this to pass it to the openLeftPanel, if we don't have this then when user opens the left panel, it doesn't know if it also has to render Misconfiguration/Vulnerabilities/Risk Score tabs

Reply:
Follow up on our discussion in DM:
1. First problem that we want to address is to encapsulate this logic in one place because right now it's copied over in 6 different places. We can start with one hook per piece of business logic (alerts, misconfigs, vulnerabilities, risk score) exposing and consuming only the pieces needed to a particular component. another possibility is to share this state through context
2. Another question, probably post 8.17 FF is to check if it's a good idea to have all these props passed to `openLeftPanel` . We do similar things in 3 places: load data we don't need in the component only to pass it to openLeftPanel. Maybe it should be self aware and get the information required to show the tabs by itself instead of getting them from the props

Reply:
Here is the followup [PR](https://github.com/elastic/kibana/pull/200291) point number 1 above